theano/tensor/opt.py

```diff
@@ -5271,6 +5271,16 @@ def local_intdiv_by_one(node):
             return [node.inputs[0].astype(node.outputs[0].dtype)]
 
 
+@register_canonicalize
+@gof.local_optimizer([T.int_div, T.true_div, T.floor_div])
+def local_zero_div(node):
+    """0 / x -> 0
+    """
+    if node.op in [T.int_div, T.true_div, T.floor_div]:
```

Comment:
That isn't good. There is some corner cases when the op is another instance. use:
isinstance(node.op, T.Elemwise) and isinstance(node.op.scalar_op, (theano.scalar.IntDiv, ...))

instead.